src/node/mempool_persist.cpp

```diff
@@ -177,12 +178,15 @@ bool DumpMempool(const CTxMemPool& pool, const fs::path& dump_path, FopenFn mock
         const uint64_t version{pool.m_opts.persist_v1_dat ? MEMPOOL_DUMP_VERSION_NO_XOR_KEY : MEMPOOL_DUMP_VERSION};
         file << version;
 
-        std::vector<std::byte> xor_key(8);
         if (!pool.m_opts.persist_v1_dat) {
+            std::vector<std::byte> xor_key(8);
             FastRandomContext{}.fillrand(xor_key);
             file << xor_key;
+            file.SetXor(xor_key);
+        } else {
+            const std::vector xor_key(8, std::byte{'\000'});
+            file.SetXor(xor_key);
         }
-        file.SetXor(xor_key);
```

Comment:
In 4437d4379c42a7b87bd01ad5ea6c450a732f4f95:
Don't see the point of this diff, must be in preparation of another commit?